platform/darwin/src/MGLRasterSource.mm

```diff
@@ -15,88 +17,77 @@ - (instancetype)initWithRawSource:(mbgl::style::RasterSource *)rawSource NS_DESI
 
 @end
 
-@implementation MGLRasterSource
-{
+@implementation MGLRasterSource {
     std::unique_ptr<mbgl::style::RasterSource> _pendingSource;
 }
 
-- (instancetype)initWithIdentifier:(NSString *)identifier URL:(NSURL *)url tileSize:(CGFloat)tileSize
-{
-    if (self = [super initWithIdentifier:identifier]) {
-        _URL = url;
-        _tileSize = tileSize;
-        [self commonInit];
-    }
-    return self;
+- (instancetype)initWithIdentifier:(NSString *)identifier configurationURL:(NSURL *)configurationURL {
+    return [self initWithIdentifier:identifier configurationURL:configurationURL tileSize:0];
```

Comment:
Thanks, I meant to follow up on this. The style specification [says](https://www.mapbox.com/mapbox-gl-style-spec/#sources-raster-tileSize) `tileSize` is optional, defaulting to 512. However, via this code path:

https://github.com/mapbox/mapbox-gl-native/blob/27db3bbdb2b0db3c4b0d8c3bd47c9c53634aca13/src/mbgl/style/tile_source_impl.hpp#L43
https://github.com/mapbox/mapbox-gl-native/blob/27db3bbdb2b0db3c4b0d8c3bd47c9c53634aca13/src/mbgl/style/tile_source_impl.cpp#L35
https://github.com/mapbox/mapbox-gl-native/blob/27db3bbdb2b0db3c4b0d8c3bd47c9c53634aca13/src/mbgl/util/mapbox.cpp#L191

a `tileSize` of 512 should cause `@2x` to be hardcoded into the tile URL template regardless of the screen resolution, whereas any other value (including 0) would cause `{ratio}` to be used. I want the `{ratio}` behavior by default, as that optimizes for the screen resolution.